jvmhs/src/Jvmhs/Data/Code.hs

```diff
@@ -158,9 +166,551 @@ verificationTypeInfo g (B.StackMapTable s) =
 $(deriveToJSON defaultOptions{fieldLabelModifier = camelTo2 '_' . drop 3} ''ExceptionHandler)
 
-instance ToJSON (B.ByteCodeOpr B.High) where
-  toJSON _ = String "<code>"
+instance ToJSON ByteCodeOpr where
+  toJSON bopr =
+    object $ ("opc" .= byteCodeOprOpCode bopr) : case bopr of
+
+    B.ArrayLoad arrType ->
+      [ "type" .= fromArrayType arrType
+      ]
+
+    B.ArrayStore arrType ->
+      [ "type" .= fromArrayType arrType
+      ]
+
+    B.Push bconstant ->
+      getListOfPairsFromBConstant bconstant
+
+    B.Load localtype localaddr ->
+      [ "type" .= fromLocalType localtype
+      , "addr" .= localaddr
+      ]
+
+    B.Store localtype localaddr ->
+      [ "type" .= fromLocalType localtype
+      , "addr" .= localaddr
+      ]
+
+    B.BinaryOpr binopr arithmetictype ->
+      [ "opr" .= binopr
+      , "type" .= fromArithmeticType arithmetictype
+      ]
+
+    B.Neg arithmetictype ->
+      [ "type" .= fromArithmeticType arithmetictype
+      ]
+
+    B.BitOpr bitopr wordsize ->
+      [ "opr" .= bitopr
+      , "size" .= wordsize
+      ]
+
+    B.IncrLocal localaddr increment ->
+      [ "addr" .= localaddr
+      , "incr" .= increment
+      ]
+
+    B.Cast castopr ->
+      [ "opr" .= castopr
+      ]
+
+    B.CompareLongs -> []
+
+    B.CompareFloating bool wordsize ->
+      [ "gt" .= bool
+      , "size" .= wordsize
+      ]
+
+    B.If cmpOpr oneOrTwo shortRelativeRef ->
+      [ "opr" .= cmpOpr
+      , "operands" .= oneOrTwo
+      , "offset" .= shortRelativeRef
+      ]
+
+    B.IfRef bool oneOrTwo shortRelativeRef ->
+      [ "equal" .= bool
+      , "operands" .= oneOrTwo
+      , "offset" .= shortRelativeRef
+      ]
+
+    B.Goto longrelativeref ->
+      [ "target" .= longrelativeref
+      ]
+
+    B.Jsr longrelativeref ->
+      [ "target" .= longrelativeref
+      ]
+
+    B.Ret localaddr ->
+      [ "addr" .= localaddr
+      ]
+
+    B.TableSwitch longrelativeref switchtable ->
+      [ "ref" .= longrelativeref
+      , "table" .= switchtable
+      ]
+
+    B.LookupSwitch longrelativeref vector ->
+      [ "ref" .= longrelativeref
+      , "vector" .= vector
+      ]
+
+    B.Get fa (B.InClass a b) ->
+      [ "static" .= fa
+      , "class" .= B.classNameAsText a
+      , "field" .= nameAndTypeFromFieldId b
+      ]
+
+    B.Put fa (B.InClass a b) ->
+      [ "static" .= fa
+      , "class" .= B.classNameAsText a
+      , "field" .= nameAndTypeFromFieldId b
+      ]
+
+    B.Invoke invocation ->
+      getInvocationAttributes invocation
+
+    B.New ref ->
+      [ "cp_entry" .= ref
+      ]
+
+    B.NewArray arraytype ->
+      getArrayType arraytype
+
+    B.ArrayLength ->
+      []
+
+    B.Throw ->
+      []
+
+    B.CheckCast ref ->
+      [ "cp_entry" .= ref
+      ]
+
+    B.InstanceOf ref ->
+      [ "cp_entry" .= ref
+      ]
+
+    B.Monitor enter ->
+      [ "enter" .= enter
+      ]
+
+    B.MultiNewArray ref word8 ->
+      [ "cp_entry" .= ref
```

Comment:
I think this is an array type... I think this might be a bug in jvm-binary. 

https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-6.html#jvms-6.5.multianewarray

Reply:
Yes I think it is a bug. According to the documentation, there are multiple integers representing the multiple dimension values for a multi-dimensional array.
Additionally, the 'NewArray' instruction (used to allocated a 1-d array) in ByteCode.hs does not have the size of the array, only it's type.

Reply:
Created a bug report https://github.com/ucla-pls/jvm-binary/issues/3.